From c6be98fd2558c7354ac1653c2d1526c89ef80f75 Mon Sep 17 00:00:00 2001
From: Andrey Borzenkov  <arvidjaar@gmail.com>
Date: Wed, 10 Apr 2013 15:57:40 +0200
Subject: [PATCH 275/364] 	* autogen.sh: Use "-h", not "-f", to test for
 existence of symbolic 	links under grub-core/lib/libgcrypt-grub/mpi.

---
 ChangeLog  | 5 +++++
 autogen.sh | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index e45ca35..fd9c082 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2013-04-09  Andrey Borzenkov <arvidjaar@gmail.com>
+
+	* autogen.sh: Use "-h", not "-f", to test for existence of symbolic
+	links under grub-core/lib/libgcrypt-grub/mpi.
+
 2013-04-08  Vladimir Serbinenko  <phcoder@gmail.com>
 
 	Fix ia64-efi image generation on big-endian machines. Deduplicate
diff --git a/autogen.sh b/autogen.sh
index 7a4b5c8..d47650b 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -24,7 +24,7 @@ ln -s ../../../grub-core/lib/libgcrypt-grub/src/g10lib.h include/grub/gcrypt/g10
 cp -R grub-core/lib/libgcrypt/mpi/generic grub-core/lib/libgcrypt-grub/mpi/generic
 
 for x in mpi-asm-defs.h mpih-add1.c mpih-sub1.c mpih-mul1.c mpih-mul2.c mpih-mul3.c mpih-lshift.c mpih-rshift.c; do
-    if [ -f grub-core/lib/libgcrypt-grub/mpi/"$x" ]; then
+    if [ -h grub-core/lib/libgcrypt-grub/mpi/"$x" ]; then
 	rm grub-core/lib/libgcrypt-grub/mpi/"$x"
     fi
     ln -s generic/"$x" grub-core/lib/libgcrypt-grub/mpi/"$x"
-- 
1.8.1.4

